byterun/dynlink.c

```diff
@@ -127,11 +127,13 @@ static char_os * parse_ld_conf(void)
 static void open_shared_lib(char_os * name)
 {
   char_os * realname;
+  char * u8;
   void * handle;
 
   realname = caml_search_dll_in_path(&caml_shared_libs_path, name);
-  caml_gc_message(0x100, "Loading shared library %"
-                  ARCH_CHARNATSTR_PRINTF_FORMAT "\n", realname);
```

Comment:
Unrelated to the PR, but am I correct that `ARCH_CHARNATSTR_PRINTF_FORMAT` is a new 4.06 thing that escaped the big `os` rename? (Will this PR end up removing it?)

Reply:
Yes, and yes.